server/util/bazel_request/bazel_request.go

```diff
@@ -33,6 +41,89 @@ func GetInvocationID(ctx context.Context) string {
 	return iid
 }
 
+type Version struct {
+	// Major, Minor, Patch are the semver version parts. For Bazel versions like
+	// "0.X", the Major and Patch versions will be 0, and the Minor version will
+	// be "x".
+	Major, Minor, Patch int
+
+	// Suffix is the raw suffix occurring immediately after the version numbers.
+	// Example: "-pre.20221102.3"
+	Suffix string
+}
+
+// GetVersion returns the parsed Bazel version. It returns nil if no Bazel
+// version could be parsed, and in particular if the client is not bazel.
+func GetVersion(ctx context.Context) *Version {
+	rmd := GetRequestMetadata(ctx)
+	if rmd == nil {
+		return nil
+	}
+	if rmd.GetToolDetails().GetToolName() != "bazel" {
+		return nil
+	}
+	tv := rmd.GetToolDetails().GetToolVersion()
+	m := bazelVersionPattern.FindStringSubmatch(tv)
+	if len(m) == 0 {
+		return nil
+	}
+	var err error
+	b := &Version{}
+	b.Major, err = strconv.Atoi(m[bazelVersionPattern.SubexpIndex("major")])
+	if err != nil {
+		return nil
+	}
+	b.Minor, err = strconv.Atoi(m[bazelVersionPattern.SubexpIndex("minor")])
+	if err != nil {
+		return nil
+	}
+	if p := m[bazelVersionPattern.SubexpIndex("patch")]; p != "" {
+		b.Patch, err = strconv.Atoi(p)
+		if err != nil {
+			return nil
+		}
+	}
+	b.Suffix = m[bazelVersionPattern.SubexpIndex("suffix")]
+	return b
+}
+
+// IsAtLeast returns whether this bazel version is equal to or supercedes the
+// given version.
+func (v *Version) IsAtLeast(c *Version) bool {
```

Comment:
I wonder if we should add a variant of this function that accepts a string version and returns (bool, err) instead, for ease of use / readability when we actually use this?

Reply:
^ going to submit as-is, and we can decide later whether we need a util method

Reply:
SGTM